### ISSUE TITLE
Make discriminated unions merge when using getFieldset / getFieldList

### ISSUE THREAD

**Issue by lifeiscontent** [edmundhung/conform#464]
### Describe the bug and the expected behavior

it'd be nice to be able to use getFieldset and getFieldList without having to run into fields either existing or not. I currently have to merge all fields into every item in the union as optional to represent that the field exists, but this feels overly cumbersome 

### Conform version

v1.0.1

### Steps to Reproduce the Bug or Issue

using the following schema:

```ts
const schema = z.discriminatedUnion("intent", [
  z.object({ intent: z.literal("greet"), message: z.string() }),
  z.object({ intent: z.literal("goodbye") }),
]);
```

I cannot use fields.message because its not defined across all items in the union

**Comment by edmundhung** [comment]
I think this will be fixed with https://github.com/edmundhung/conform/pull/459.

**Comment by edmundhung** [comment]
This should be fixed in [v1.0.2](https://github.com/edmundhung/conform/releases/tag/v1.0.2).
